### PR TITLE
BET-5522: Workspaces support.

### PR DESCRIPTION
**Pull request by nikchern**

### CODE REVIEW

File: examples/resources/spectrocloud_workspace/resource.tf [hunk 1]
```diff
@@ -0,0 +1,80 @@
+data "spectrocloud_cluster" cluster1 {
+  name = "vsphere-picard-2"
+}
+
+resource "spectrocloud_workspace" "workspace" {
+  name = "wsp-tf"
+
+  clusters {
+    name = data.spectrocloud_cluster.cluster1.name
+    uid = data.spectrocloud_cluster.cluster1.id
```
Comment (rishi-anand) [inline examples/resources/spectrocloud_workspace/resource.tf:10]: @nikchern, Should we just take uid as input?

File: examples/resources/spectrocloud_workspace/resource.tf [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+    subjects {
+      type      = "ServiceAccount"
+      name      = "testrolesubject3"
+      namespace = "testrolenamespace"
+    }
+  }
+
+  namespaces {
+    name = "test5ns"
```
Comment (rishi-anand) [inline examples/resources/spectrocloud_workspace/resource.tf:58]: Just a heads up for `IsNamespaceRegex` bool will be added into the api in 2.7 itself

File: spectrocloud/resource_workspace.go [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+		}
+		if d.HasChange("namespaces") {
+			if err := c.UpdateWorkspaceResourceAllocation(d.Id(), toUpdateWorkspaceNamespaces(d)); err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+
+	/*
+		if d.HasChange("backup_policy") {
```
Comment (rishi-anand) [inline spectrocloud/resource_workspace.go:259]: @nikchern we will need to check for `d.HasChange("backup_policy")`

**Review by rishi-anand — APPROVED** [review]

File: examples/resources/spectrocloud_workspace/resource.tf (hunk 1 above)
Comment (nikchern) [reply examples/resources/spectrocloud_workspace/resource.tf:10]: @rishi-anand filed a bug to support it, as of now both needed.

File: examples/resources/spectrocloud_workspace/resource.tf (hunk 2 above)
Comment (nikchern) [reply examples/resources/spectrocloud_workspace/resource.tf:58]: Adding support for regex

File: spectrocloud/resource_workspace.go (hunk 3 above)
Comment (nikchern) [reply spectrocloud/resource_workspace.go]: Implemented.

**Pull request merged**